### DISCUSSION TITLE
icon 的 class 用变量引用就无法显示

### DISCUSSION THREAD

**Discussion by vinciarts** [unocss/unocss#4002] [Q&A]
这样写可以正常显示：
`<span class="i-bxl-apple text-2xl"></span>`

但是这样就无法显示：
`<span class="{icon} text-2xl"></span>`

`{icon}` 是从外部导入的。除非我先按前者写，然后 icon 在缓存里，才能显示。但生产环境不能这样…

**Comment by pdsuwwz** [comment]
Hello, This is normal and designed like this. For details：https://unocss.dev/guide/extracting#safelist

![image](https://github.com/user-attachments/assets/e9dd2598-684c-4e7f-8f1f-bd7dda59d953)
